src/liboslexec/llvm_ops.cpp

```diff
@@ -772,11 +772,37 @@ OSL_SHADEOP void osl_filterwidth_vdv(void *out, void *x_)
 
 
 
-
-
 // Asked if the raytype includes a bit pattern.
 OSL_SHADEOP int osl_raytype_bit (void *sg_, int bit)
 {
     ShaderGlobals *sg = (ShaderGlobals *)sg_;
     return (sg->raytype & bit) != 0;
 }
+
+
+
+// extern declaration
+OSL_SHADEOP int osl_range_check_err (int indexvalue, int length,
+                         const char *symname, void *sg,
+                         const void *sourcefile, int sourceline,
+                         const char *groupname, int layer,
+                         const char *layername, const char *shadername);
+
+
+
+OSL_SHADEOP int
+osl_range_check (int indexvalue, int length, const char *symname,
+                 void *sg, const void *sourcefile, int sourceline,
+                 const char *groupname, int layer, const char *layername,
+                 const char *shadername)
+{
+    if (indexvalue < 0 || indexvalue >= length) {
```

Comment:
Laurent was benchmarking a particular shader in isolation (with testshade, not in a render) that went from 7.8 to 6.4 when range checking was disabled entirely. This patch lowers the cost to 7.1 (i.e. eliminating roughly half of the added expense of the range check). 

I tried two production scenes. One appears to speed up by about 1% overall, the other did not have a measurable difference.